### PR TITLE
chore: copy document catalogue when truncating db

### PR DESCRIPTION
**Pull request by arriqaaq**

### CODE REVIEW

File: embedded/document/engine.go [hunk 1]
```diff
@@ -978,3 +978,8 @@ func (d *documentReader) Read(ctx context.Context, count int) ([]*structpb.Struc
 func (d *documentReader) Close() error {
 	return d.reader.Close()
 }
+
+// CopyCatalogToTx copies the current sql catalog to the ongoing transaction.
+func (e *Engine) CopyCatalogToTx(ctx context.Context, tx *store.OngoingTx) error {
+	return e.sqlEngine.CopyCatalogToTx(ctx, tx)
```
Comment (jeroiraz) [inline embedded/document/engine.go:984]: isn't it covered with the sql catalog already?

note: is the sql catalog copied with the indexing up to date?

File: pkg/database/database.go [hunk 2]
```diff
@@ -1656,6 +1656,17 @@ func logErr(log logger.Logger, formattedMessage string, err error) error {
...
-	// TODO: add document store support for truncation too
-	return d.sqlEngine.CopyCatalogToTx(ctx, tx)
+	// copy the sql catalog
+	err := d.sqlEngine.CopyCatalogToTx(ctx, tx)
+	if err != nil {
+		return err
+	}
+
+	// copy the document store catalog
+	err = d.documentEngine.CopyCatalogToTx(ctx, tx)
```
Comment (jeroiraz) [inline pkg/database/database.go:1666]: may not be needed... ?

File: embedded/document/engine.go (hunk 1 above)
Comment (arriqaaq) [reply embedded/document/engine.go:984]: document sql engine is instantiated with a different prefix

File: pkg/database/database.go (hunk 2 above)
Comment (arriqaaq) [reply pkg/database/database.go:1666]: document sql engine is instantiated with a different prefix

**Review by jeroiraz — APPROVED** [review]

**Pull request merged**